### PR TITLE
Orion/push package

### PR DESCRIPTION
**Pull request by borland**

### CODE REVIEW

File: pkg/cmd/package/upload/upload.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/packages"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// TODO Outstanding questions for package upload:
+// - What should the response be for outputformat basic and json?
```
Comment (jbristowe) [inline pkg/cmd/package/upload/upload.go:22]: The basic output from `octo` looks something like this:

```
Octopus CLI, version 9.x.y

Detected automation environment: "NoneOrUnknown"
Space name unspecified, process will run in the default space context
Handshaking with Octopus Server: https://foobar.octopus.app
Handshake successful. Octopus version: 2022.x.y; API version: 3.0.0
Authenticated as: Foo Bar <foo@bar.com> 
Pushing package: /Users/foobar/foo.1.1.0.zip...
Requesting signature for delta compression from the server for upload of a package with id 'foo' and version '1.1.0'
No package with the same ID exists on the server
Falling back to pushing the complete package to the server
Push successful
```

Here's the output in JSON:

```
{
  "SuccessfulPackages": [
    "/Users/johnbristowe/GitHub/OctopusCLI/foo.1.1.1.zip"
  ],
  "FailedPackages": []
}
```

File: pkg/cmd/package/upload/upload.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+// - Basic output: I have "Successfully uploaded package x-1.0.0.zip" when the package was created/overwritten, and "Successfully processed package x-1.0.0.zip" when it was ignored. Is this right?
```
Comment (jbristowe) [inline pkg/cmd/package/upload/upload.go:23]: `octo` will inform you if the package is NOT overwritten. We should do the same.

File: pkg/cmd/package/upload/upload.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+// - If we specify multiple packages and one of them fails, does it keep going or abort on first error?
```
Comment (jbristowe) [inline pkg/cmd/package/upload/upload.go:24]: `octo` reports successful and failed packages (for upload). We could do the same.

File: pkg/cmd/package/upload/upload.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+// - the package upload response from the server has a load of stuff in it. Is there any value in doing anything with this?
```
Comment (jbristowe) [inline pkg/cmd/package/upload/upload.go:25]: Not at the moment. But perhaps in the future?

File: pkg/cmd/package/upload/upload.go (hunk 2 above)
Comment (borland) [reply pkg/cmd/package/upload/upload.go:23]: ok cool I'll have a play with `octo` and put in some similar wording

File: pkg/cmd/package/upload/upload.go (hunk 3 above)
Comment (borland) [reply pkg/cmd/package/upload/upload.go:24]: cool, I will keep going

File: pkg/cmd/package/upload/upload.go (hunk 4 above)
Comment (borland) [reply pkg/cmd/package/upload/upload.go:25]: Cool, thanks for that. I'll remove this TODO comment and add another one at the callsite explaining that while `UploadPackageResponse` has a lot of data in it, the CLI deliberately doesn't use any at this time

File: pkg/cmd/package/upload/upload.go (hunk 1 above)
Comment (borland) [reply pkg/cmd/package/upload/upload.go:22]: My intent with all these TODOS was to merge this PR first as a 'first pass' that people could play with, then do a second pass and tidy them all up, plus any other UX/wording feedback we gathered.

However, I didn't actually tell anyone about that. Whoops, so of course you'd reject the PR on grounds of unresolved TODOs :-)

**Review by domenicsim1 — APPROVED** [review]

**Pull request merged**